src/util/simplify_expr2.cpp

```diff
@@ -890,6 +890,26 @@ expr2tc member2t::do_simplify() const
 
     return s;
   }
+  else if (is_with2t(source_value))
+  {
+    auto current_source = source_value;
+
+    // Traverse through nested 'with' expressions
+    while (is_with2t(current_source))
+    {
+      const auto &with = to_with2t(current_source);
+
+      // Check if the member matches the update field
+      if (member == to_constant_string2t(with.update_field).value)
```

Comment:
is `with.update_field` always a constant_string? If it isn't, you need to check it before calling `to_constant_string`

Reply:
It is contant_string.

Reply:
We don't have this constraint written anywhere, `update_field` is declared as an `expr2tc`, so let's add a check here.